Add `VO:0005505` - vaccine for platform

cthoyt opened the pull request:
Part of #619.

Similarly to #620, this adds a mid-level grouping term between VO:0000001 (vaccine) and several children representing vaccine platforms:

1. conjugate vaccine
2. DNA vaccine
3. inactivated vaccine
4. live virulent vaccine
5. live attenuated vaccine
6. passive vaccine
7. toxoid vaccine
8. whole organism vaccine
9. virus-like particle vaccine
10. recombinant vector vaccine
11. RNA vaccine
12. subunit vaccine

This is added in the term "vaccine for platform" (VO:0005505). This doesn't add any axiomization regarding this grouping at the moment, since it's not yet clear if there's a shared way of ontologizing "what is a platform". Looking forward, platforms are typically defined as a combination of the one or more of the components/antigens/adjuvants/other ingredient types

Note: this PR is fully independent of #622

cthoyt commented:
cc @yongqunh @Huffmaar

Huffmaar commented:
I currently would raise the potential point that some of these categories
would cause overlap.
1. 'whole organism vaccine' has possible overlap with live virulent or live
attenuated vaccines as the live attenuated vaccines can be composed of
whole organism vaccines that have undergone some process while remaining
attenuated. However, depending on what definition is used for 'whole
organism vaccine', it would be possible to have an attenuated strain that
is missing or has altered a specific virulence factor; would that still
meet the implicit definition of whole organism vaccine.
 2. There is a specific category of vaccines known as cocktail vaccines,
which are composed of two or more distinct vaccines (VO_0001395), including
vaccines from multiple categories (e.g. cocktail vaccine containing subunit
vaccine and inactivated whole organism vaccine (VO_0001396)).
3.  I am not sure I am fully on board with your use and definition of a
platform. Right now these are recognized as distinct types of vaccines,
which I would classify based on their construction. I typically think of
platforms as something as part of an Illumina chips, but will think it over
tonight.
4. Otherwise most of these should have an axiom of 'has role' some
['vaccine role'].

On Sun, Sep 24, 2023 at 1:14 PM Charles Tapley Hoyt <
***@***.***> wrote:

> cc @yongqunh <https://github.com/yongqunh> @Huffmaar
> <https://github.com/Huffmaar>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/vaccineontology/VO/pull/623#issuecomment-1732623098>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANDAISGIX5WOA2EP2SAYTZTX4BS7ZANCNFSM6AAAAAA5FB7HOA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Anthony Huffman,
DCMB Student,
He Lab

cthoyt commented:
1. I think this is okay, since you could make subclasses that use both of those as mixins
2. I don't think that combining multiple vaccines together constitutes as a platform itself - the platform is more about the combination of ingredients that make up a single one. There might also be a different BFO branch under which vaccine cocktails should go that's different from vaccines at all
3. A lot of recent literature uses the phrase "vaccine platform" (a few examples: https://ncirs.org.au/vaccine-platforms, https://www.belfercenter.org/publication/technology-factsheet-vaccine-platforms, https://www.sciencedirect.com/science/article/pii/S2001037021001732). I think it means the same thing as what you are saying - barring that Illumina also uses "platform" to mean something different in their domain. If you don't like "vaccine for platform" as a term, I am happy to also change this to "vaccine by type" or something else
4. Does this mean you want me to add that as a top-level axiom for the "vaccine for platform" term?

Huffmaar commented:
1. Then it should be fine.
3. Alright, in that case. I would ask add a synonym; 'vaccine by platform type'. 
4. That is my current preference, but want to confer with @yongqunh before committing. If need be, there are specific children of 'vaccine role' that can be added. 'live attenuated vaccine' and 'inactivated vaccine' would not have a role, but inherit 'whole vaccine role' from 'whole vaccine'.

I'm still worried about 2 and it showing up due to shared inheritances (i.e. which one would be primary when we add new vaccines). Especially as cocktail vaccines can either share the same platform, or differ (usually as part of prime-boost vaccines).  But other than that, I don't have any further issues.

cthoyt commented:
hi @Huffmaar, I updated the name based on your suggestion and rebased on the main branch. 

If I were encoding a cocktail vaccine, I would make relations that it has a component with multiple other vaccines, each of which can have the platforms annotated on them as subclasses directly

Huffmaar commented:
Great, I'll look at those later today. Also, right now I would ask that
tag @zhengj2007 since she is now the main person who is consolidating
changes on VO now.

I would ask what explicitly what relation would be defined so that it works
with the reasoner.

On Wed, Nov 15, 2023 at 5:58 AM Charles Tapley Hoyt <
***@***.***> wrote:

> hi @Huffmaar <https://github.com/Huffmaar>, I updated the name based on
> your suggestion and rebased on the main branch.
>
> If I were encoding a cocktail vaccine, I would make relations that it has
> a component with multiple other vaccines, each of which can have the
> platforms annotated on them as subclasses directly
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/vaccineontology/VO/pull/623#issuecomment-1812281500>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANDAISCAZUSIBEIYWNVFN73YESN6XAVCNFSM6AAAAAA5FB7HOCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQMJSGI4DCNJQGA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Anthony Huffman,
DCMB Student,
He Lab

cthoyt commented:
I think the discussion about the components should be separated from this PR, but vo:0000520 might work
